backend/src/main/kotlin/org/loculus/backend/controller/FilesController.kt

```diff
@@ -23,8 +33,44 @@ class FilesController(
     private val filesDatabaseService: FilesDatabaseService,
     private val s3Service: S3Service,
     private val groupManagementPreconditionValidator: GroupManagementPreconditionValidator,
+    private val submissionDatabaseService: SubmissionDatabaseService,
+    private val accessionPreconditionValidator: AccessionPreconditionValidator,
 ) {
 
+    @GetMapping("/get/{accession}/{version}/{fileField}/{fileName}")
+    fun getFileDownloadUrl(
+        @HiddenParam user: User,
+        @PathVariable accession: Accession,
+        @PathVariable version: Long,
+        @PathVariable fileField: String,
+        @PathVariable fileName: String,
+    ): ResponseEntity<Void> {
+        val accessionVersion = AccessionVersion(accession, version)
+        val fileId = submissionDatabaseService.getFileId(accessionVersion, fileField, fileName)
+        if (fileId == null) {
+            throw NotFoundException("File not found")
+        }
+        val isPublic = filesDatabaseService.isFilePublic(fileId)!!
+        if (!isPublic) {
+            if (user is AuthenticatedUser) {
+                accessionPreconditionValidator.validate {
+                    thatAccessionVersionExists(accessionVersion)
+                        .andThatUserIsAllowedToEditSequenceEntries(user)
+                }
+            } else {
+                throw ForbiddenException("Authentication needed: the file is not public")
+            }
+        }
+        val presignedUrl = s3Service.createUrlToReadPrivateFile(
+            fileId,
+            filesDatabaseService.getGroupId(fileId)!!,
+            fileName,
+        )
```

Comment:
We create a presignedUrl, even if the file is public here, right? We don't need to do that, right? IMO it would be cool to just link to the file normally if it is public, that means the link is valid for longer.

Reply:
> We create a presignedUrl, even if the file is public here, right? We don't need to do that, right?

We need to so it gets the correct filename

Reply:
Ah yes, you're right, the link will be valid.

Reply:
(Agree with what Theo said)